contracts/random/random.go

```diff
@@ -69,10 +69,6 @@ func RevealAndCommit(vmRunner vm.EVMRunner, randomness, newCommitment common.Has
 	log.Trace("Revealing and committing randomness", "randomness", randomness.Hex(), "commitment", newCommitment.Hex())
 	err := revealAndCommitMethod.Execute(vmRunner, nil, common.Big0, randomness, newCommitment, proposer)
 
-	// FIXME this should be done by the caller, remove from here after
-	if err == nil {
-		vmRunner.GetStateDB().Finalise(true)
```

Comment:
why are you removing this? It's not clear to me that this can be removed

Reply:
They are already handled, check my comment:
https://github.com/celo-org/celo-blockchain/pull/1573#discussion_r656619294

Reply:
nice